### PR TITLE
Feat/add calendar icon position support

### LINKED ISSUE #451: feat: add `calendarIconPosition` prop in DatePicker
### Description
The current Medly DatePicker component shows the Calendar icon on the right side. As a user, I want the ability to specify where I want to show the Calendar icon.

image

### Solution you'd like
Expose a prop called `calendarIconPosition` in the DatePicker component which will accept only two values - "left" or "right".
Based on the value of this prop, the location of the Calendar icon will be shown.

### Additional context
- The `calendarIconPosition` prop should be optional and have the default value of "right".
- Add detailed documentation in layman language.
- Add a working example demonstrating the usage of `calendarIconPosition` prop on with both `right` and `left` values.
- A sample unit test for testing the new prop. (good to have)

### PR DESCRIPTION
**Pull request by rayng86**
### PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] Tests for the changes have been added (for bug fixes/features)
-   [x] Docs have been added/updated (for bug fixes/features)

### PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [x] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [ ] Performance improves
-   [ ] Adding missing tests
-   [ ] Documentation content changes
-   [ ] Other... Please describe:

### What is the current behavior?

calendar icon currently supports right alignment

### What is the new behavior?
 #451  - add a new prop called calendarIconPosition that will allow a user to set and position the calendar icon left or right. 

### Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### Additional context

Provide any additional context or screenshots about the feature PR.

https://user-images.githubusercontent.com/74542/135578313-ffcd21b5-d226-495a-9bbf-acc5d2fe8809.mp4

image

### CODE REVIEW

**Comment by gmukul01** [comment]
@rayng86  Can you please add basic unit test also ?

**Comment by rayng86** [comment]
> @rayng86 Can you please add basic unit test also ?

@gmukul01 Sure, added tests

File: packages/core/src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -144,7 +145,7 @@ const Component: React.FC<DatePickerProps> = React.memo(
                     key={inputKey}
                     ref={inputRef}
                     required={required}
-                    {...(showCalendarIcon && { suffix: suffixEl })}
+                    {...(showCalendarIcon && (calendarIconPosition === 'left' ? { prefix: dateIconEl } : { suffix: dateIconEl }))}
```
Comment (wsameer) [inline packages/core/src/components/DatePicker/DatePicker.tsx:148]: looks good

**Review by gmukul01 — APPROVED** [review]
LGTM

**Review by wsameer — APPROVED** [review]
